### PR TITLE
Merge #14679: importmulti: Don't add internal addresses to address book

### PR DESCRIPTION
**Pull request by ogabrielides**
bitcoin#14679

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
Adjusted the name, please use this naming scheme ( / likely backport with -m1 as described in #4000)

File: src/wallet/rpcdump.cpp [hunk 1]
```diff
@@ -1284,7 +1284,8 @@ static UniValue ProcessImport(CWallet * const pwallet, const UniValue& data, con
                 }
 
                 // add to address book or update label
```
Comment (PastaPastaPasta) [inline src/wallet/rpcdump.cpp:1286]: this comment is supposed to be updated
```suggestion
                // if not internal add to address book or update label
```

File: src/wallet/rpcdump.cpp [hunk 2]
```diff
@@ -1284,7 +1284,8 @@ static UniValue ProcessImport(CWallet * const pwallet, const UniValue& data, con
...
-                if (IsValidDestination(dest)) {
+                if(!internal) {
+                    assert(IsValidDestination(dest));
                     pwallet->SetAddressBook(dest, label, "receive");
                 }
```
Comment (PastaPastaPasta) [inline src/wallet/rpcdump.cpp:1291]: Missing the changes here 

https://github.com/bitcoin/bitcoin/commit/f617e05c38c9c7e663b824fefe5329cc5cbbf26a#diff-c455c7835ec6d7a6d846c1ce3d3c2ad704b01d0c2215bffb5c28c9a5a1b33369L1105-R1106

File: test/functional/wallet_importmulti.py [hunk 3]
```diff
@@ -101,16 +103,16 @@ def run_test (self):
         assert_equal(address_assert['ismine'], False)
         assert_equal('timestamp' in address_assert, False)
 
-
-        # Address + Public key + !Internal
+        # Address + Public key + !Internal(explicit)
```
Comment (PastaPastaPasta) [inline test/functional/wallet_importmulti.py:106]: Don't delete this line
```suggestion

        # Address + Public key + !Internal(explicit)
```

File: test/functional/wallet_importmulti.py [hunk 4]
```diff
@@ -135,6 +137,7 @@ def run_test (self):
         assert_equal(address_assert['ismine'], False)
         assert_equal(address_assert['timestamp'], timestamp)
 
+
```
Comment (PastaPastaPasta) [inline test/functional/wallet_importmulti.py:140]: Don't add this line
```suggestion
```

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

**Comment by ogabrielides** [comment]
@PastaPastaPasta Updates completed

File: test/functional/wallet_importmulti.py [hunk 5]
```diff
@@ -43,7 +43,7 @@ def run_test (self):
 
         # RPC importmulti -----------------------------------------------
 
-        # Bitcoin Address
+        # Bitcoin Address(implicit non-internal)
```
Comment (UdjinM6) [inline test/functional/wallet_importmulti.py:46]:
```suggestion
        # Bitcoin Address (implicit non-internal)
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — DISMISSED** [review]
utACK

File: src/wallet/rpcdump.cpp [hunk 6]
```diff
@@ -1283,8 +1283,9 @@ static UniValue ProcessImport(CWallet * const pwallet, const UniValue& data, con
                     throw JSONRPCError(RPC_WALLET_ERROR, "Error adding address to wallet");
                 }
 
-                // add to address book or update label
-                if (IsValidDestination(dest)) {
+                // if not internal add to address book or update label
+                if(!internal) {
```
Comment (PastaPastaPasta) [inline src/wallet/rpcdump.cpp:1287]:
```suggestion
                if (!internal) {
```

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**